W-12431733-Exclude-Studio-from-Win-Defender-CP (#451)
mulesoft/docs-studio: modules/ROOT/pages/to-download-and-install-studio-wx.adoc

. Download the Anypoint Studio installation file from the https://www.mulesoft.com/lp/dl/studio[Download Site].
. Extract the Windows installation file into the `C:\` root folder.
** Ensure that your user has read and write permissions for the `C:\` directory.
** If you are using a Windows antivirus application, ensure that the `plugins/` and `features/` directories are under the *trusted* category.
. Open the extracted Anypoint Studio.
+
When Studio first runs, the embedded OpenJDK Platform libraries might cause a Windows Security alert. +
Ensure to allow the OpenJDK Platform library from Temurin publisher through your Windows firewall.
. Click *OK* to accept the default workspace. +
** If you choose to use an existing workspace, this must have been created using Anypoint Studio 7.0 or later. +
Studio 7.x supports only Mule 4.x projects, because the structure of the project, export format, XML, and scripting language are different than the ones used by Mule 3.x.

== Exclude Studio from Windows Defender Scans

> While Windows Defender is a tool for keeping your computer safe, it can also slow down the performance of Studio because it scans files in real-time as they are accessed, which causes delays. To prevent performance issues, you can exclude Studio from Windows Defender scans. 

Reviewer: you can **completely** exclude Studio from Windows Defender scans. 

I would add the word  **completely**  because the first option in this document excludes the folders `plugins/` and `features/`.

Author: Great catch! Thanks!

> . Run Studio. The following window pop-ups to ask if Studio folders and processes should be excluded:

Reviewer: After running Studio_, if Windows Defender Real Time Scanning is enabled,_ the following window pop-ups to ask if Studio folders and processes should be excluded. 

Windows Defender is not the only real time scanning virus tool the user can has, and this feature only works with Windows Defender. i.e. Salesforce machines has Falcon, so this feature doesn't work for that windows machines.

Reviewer: Also, I would add an explanation of what each folder is about:

- /.p2 Stores all Studio/Eclipse plugins. 
- The workspace the user selects when launched Studio. 
- The entire Anypoint Studio application folder.

Author: Thanks for the clarification! 
It may be misleading for the reader if we don't mention "Windows Defender" in the first step.

Author: Made some changes! 
Let's see if they work! :)

+
image::windows-defender-exclusions.png[“Windows Defender exclusions popup asks the user if folder and processes should be excluded.”]
+

> . Select the *Remember my decision* checkbox.

Reviewer: This is optional, if users select this and then selects No, this popup would never open again for this workspace.

Author: If the user does not select the "Remember my decision" checkbox, the popup will appear when other workspaces are created in Studio? Did I get you right?

Reviewer: it always be shown in new workspaces, but if you click remember my decision, and **No** was selected, the pop-up won't show anymore for that workspace.

Author: I've made some changes. Let's see if they work! :)

. Extract the Windows installation file into the `C:\` root folder.
** Ensure that your user has read and write permissions for the `C:\` directory.
** If you are using a Windows antivirus application, ensure that the `plugins/` and `features/` directories are under the *trusted* category.
. Open the extracted Anypoint Studio.
+
When Studio first runs, the embedded OpenJDK Platform libraries might cause a Windows Security alert. +
Ensure to allow the OpenJDK Platform library from Temurin publisher through your Windows firewall.
. Click *OK* to accept the default workspace. +
** If you choose to use an existing workspace, this must have been created using Anypoint Studio 7.0 or later. +
Studio 7.x supports only Mule 4.x projects, because the structure of the project, export format, XML, and scripting language are different than the ones used by Mule 3.x.

== Exclude Studio from Windows Defender Scans

While Windows Defender is a tool for keeping your computer safe, it can also slow down the performance of Studio because it scans files in real-time as they are accessed, which causes delays. To prevent performance issues, you can exclude Studio from Windows Defender scans. 

> . Configure Windows Defender to scan your machine in real-time.

Reviewer: This is not a necessary step, if windows defender real-time scanning is enabled, the pop-up would show.

Author: OK! Thanks for the data!

. Run Studio. The following window pop-ups to ask if Studio folders and processes should be excluded:
+

> image::windows-defender-exclusions.png[“Windows Defender exclusions popup asks the user if folder and processes should be excluded.”]

Reviewer: The pop-up image should be changed by another, it might be taken from the RC.

Author: @mauriarroqui, can you share the popup image from RC when available?
If not, who may help me with this SS? Thanks!

Author: Thanks for the SS! ;)

. Download the Anypoint Studio installation file from the https://www.mulesoft.com/lp/dl/studio[Download Site].
. Extract the Windows installation file into the `C:\` root folder.
** Ensure that your user has read and write permissions for the `C:\` directory.

> ** If you are using a Windows antivirus application, ensure that the `plugins/` and `features/` directories are under the *trusted* category.

Reviewer 2: ```suggestion
** If you use a Windows antivirus application, ensure that the `plugins/` and `features/` directories are under in the *trusted* category.
```

. Open the extracted Anypoint Studio.
+
When Studio first runs, the embedded OpenJDK Platform libraries might cause a Windows Security alert. +
Ensure to allow the OpenJDK Platform library from Temurin publisher through your Windows firewall.
. Click *OK* to accept the default workspace. +
** If you choose to use an existing workspace, this must have been created using Anypoint Studio 7.0 or later. +
Studio 7.x supports only Mule 4.x projects, because the structure of the project, export format, XML, and scripting language are different than the ones used by Mule 3.x.

== Exclude Studio from Windows Defender Scans

> While Windows Defender is a tool for keeping your computer safe, it can also slow down the performance of Studio because it scans files in real-time as they are accessed, which causes delays. To prevent performance issues, you can completely exclude Studio from Windows Defender real-time protection.

Reviewer 2: Can you remove the word "completely"? I don't think it's necessary.

Author: This is something we discussed with Mauricio, the dev (https://github.com/mulesoft/docs-studio/pull/451#discussion_r1120248802).
As there are several "things" of Studio that can be excluded from Windows Defender scans, we added the "completely" so that it is clear that all the Studio "things" should be excluded.
Would "completely" work in this case? Thanks!

> While Windows Defender is a tool for keeping your computer safe, it can also slow down the performance of Studio because it scans files in real-time as they are accessed, which causes delays. To prevent performance issues, you can completely exclude Studio from Windows Defender real-time protection.

Reviewer 2: ```suggestion
While Windows Defender is a tool for keeping your computer safe, it can slow down the performance of Studio because it scans files in real-time as they are accessed, which causes delays. To prevent performance issues, you can exclude Studio from Windows Defender real-time protection:
```

> . Run Studio. The following Windows Defender real-time protection notice pop-ups to ask if Studio folders and processes should be excluded:

Reviewer 2: ```suggestion
. Run Studio. 
+
A Windows Defender real-time protection notice asks whether to exclude Studio folders and processes:
```

Reviewer 2: I don't think you need "the following" in this context.

Author: I like it your way too! Thanks!

+

> image::windows-defender-exclusions.png[“Windows Defender real-time protection exclusions popup asks the user if folder and processes should be excluded.”]

Reviewer 2: ```suggestion
image::windows-defender-exclusions.png[“A Windows Defender real-time protection popup asks whether to exclude Studio folders and processes.”]
```

+

> * */.p2* folder stores Studio plugins.

Reviewer 2: ```suggestion
* The `/.p2` folder stores Studio plugins.
```

+

> * The second folder shows the path of the workspace the user selects when launching Studio.

Reviewer 2: ```suggestion
* The second folder shows the path of the workspace that you select when launching Studio.
```

Reviewer 2: Maybe "the user" is correct in this context. If so, ignore my edit.

Author: Your comment is correct! Thanks!

+

> * The third folder point to Anypoint Studio application location.

Reviewer 2: ```suggestion
* The third folder points to the Anypoint Studio application location.
```

+
. Optionally, select the *Remember my decision* checkbox.
. Click *Yes*.

> . Click *Yes* again when Windows PowerShell requests permission to make the change. 

Reviewer 2: ```suggestion
. Click *Yes* again when Windows PowerShell requests permission to make the change:
```

+

> image::windows-powershell-access-request.png[“Windows Powershell popup requests user to confirm the change.”]

Reviewer 2: ```suggestion
image::windows-powershell-access-request.png[“Windows Powershell popup requests that you confirm the change.”]
```

> NOTE: If you select the *Remember my decision* checkbox and click *No*, Windows Defender real-time protection scans Studio folders and processes as you work with it.

Reviewer 2: processes what?

> NOTE: If you select the *Remember my decision* checkbox and click *No*, Windows Defender real-time protection scans Studio folders and processes as you work with it.

Reviewer 2: "with it" is not clear. Does this mean as you work with Studio or as you work with Windows Defender? If the former - "as you work with them."

Author: Great catch! Thanks!
